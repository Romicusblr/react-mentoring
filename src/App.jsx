import React from 'react';
import {
  BrowserRouter as Router, Route, Switch, Redirect,
} from 'react-router-dom';
import MainPage from './pages/MainPage';
import MoviePage from './pages/MoviePage';
import Layout from './pages/Layout';

const App = () => (
  <Layout>
    <Switch>
      <Redirect exact from="/" to="/search" />
      <Route path="/search" component={MainPage} />
      <Route path="/film/:id" component={MoviePage} />
      <Route component={() => <div>Not Found</div>} />
    </Switch>
  </Layout>
);

export default App;
