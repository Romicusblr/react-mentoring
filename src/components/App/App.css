* {
  color: #FFF;
  font-size: 16px;
}

body {
  background-color: #545454
}
