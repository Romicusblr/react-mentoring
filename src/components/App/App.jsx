import React from 'react';
import Header from '../Header';
// import HeaderMovieDetails from '../HeaderMovieDetails';
import Main from '../Main';
import Footer from '../Footer';
import ErrorBoundary from '../ErrorBoundary';
import './App.css';
import fakeData from '../fakeMovies';

const App = () => (
  <React.Fragment>
    <ErrorBoundary>
      <Header />
      {/* <HeaderMovieDetails movie={fakeData.data[0]} /> */}
      <Main data={fakeData} />
      <Footer />
    </ErrorBoundary>
  </React.Fragment>
);

export default App;
