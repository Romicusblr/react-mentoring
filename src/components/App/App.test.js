import App from '.';

describe('App', () => {
  it('should match to snapshot', () => {
    const snap = shallow(<App />);
    expect(snap).toMatchSnapshot();
  });
});
