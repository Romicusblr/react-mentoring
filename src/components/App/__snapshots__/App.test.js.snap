// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match to snapshot 1`] = `
<Fragment>
  <ErrorBoundary>
    <Header
      className=""
    />
    <Connect(MainContainer) />
    <Footer
      className=""
    />
  </ErrorBoundary>
</Fragment>
`;
